### ISSUE TITLE
Authenticating with TokenCredentials throws exception

### ISSUE THREAD

**Issue by gliljas** [Azure/azure-libraries-for-net#249]
I've tried to authenticate using a token, but an exception is thrown.

```
var client = RestClient
    .Configure()
    .WithEnvironment(AzureEnvironment.AzureGlobalCloud)
    .WithLogLevel(HttpLoggingDelegatingHandler.Level.Basic)
    .WithCredentials(new TokenCredentials(token))
    .Build();


var azure = Microsoft.Azure.Management.Fluent.Azure
    .Authenticate(client, directoryId)
    .WithSubscription(subscriptionId);
```

which results in 

```Exception thrown: 'System.InvalidCastException' in Microsoft.Azure.Management.Compute.Fluent.dll
An unhandled exception of type 'System.InvalidCastException' occurred in System.Private.CoreLib.dll
Unable to cast object of type 'Microsoft.Rest.TokenCredentials' to type 'Microsoft.Azure.Management.ResourceManager.Fluent.Authentication.AzureCredentials'.```

**Comment by gliljas** [comment]
While I've worked around the problem, it's still an ugly exception.

**Comment by hovsepm** [comment]
yeah, that is already addressed in PR https://github.com/Azure/azure-libraries-for-net/pull/195 .

**Comment by gliljas** [comment]
Super. Thanks!

**Comment by prerakpradhan** [comment]
@gliljas looks like the change has not yet been released and I am hitting this issue now. What was your work around for this issue?

**Comment by hovsepm** [comment]
PR #195 is merged. Fix will be shipped in v1.9
